experimental/dds/tree2/src/core/tree/cursor.ts

```diff
@@ -363,6 +363,36 @@ export function forEachNode<TCursor extends ITreeCursor = ITreeCursor>(
 	}
 }
 
+/**
+ * Moves cursor to each child of the current field.  The optional arguments `start` and `end`
+ * can be used to restrict the range of children visited.
+ *
+ * `callbackFn` is invoked after moving the cursor to each child.  The callback must return
+ * `true` to continue the enumeration, or `false` to early exit.
+ *
+ * The `callbackFn` may move move the provided cursor, but must restore the cursor to the current
+ * child node in the enumeration before returning.
+ */
+export function forEachNodeIn<TCursor extends ITreeCursor = ITreeCursor>(
+	cursor: TCursor,
+	callbackFn: (cursor: TCursor) => boolean,
+	start = 0,
+	end = cursor.getFieldLength(),
+): void {
+	assert(cursor.mode === CursorLocationType.Fields, 0x3bd /* should be in fields */);
+
+	// The placement of `remaining > 0` in the expressions below avoids unnecessary
+	// attempts to move the cursor when 0 items remain in the enumeration.
+	for (
+		let remaining = end - start,
+			shouldContinue = remaining > 0 && cursor.firstNode() && cursor.seekNodes(start);
+		shouldContinue;
+		shouldContinue = shouldContinue && --remaining > 0 && cursor.nextNode()
+	) {
+		shouldContinue = callbackFn(cursor) !== false;
```

Comment:
Why check your boolean is not false? We have a type system so we should trust it
```suggestion
		shouldContinue = callbackFn(cursor);
```

Reply:
Remnant from allowing the callback to return 'undefined' if not using the early exit behavior.